### ISSUE TITLE
feat(R): Implement BatchNormalization layer translation in orbital_keras_dag_impl (currently incorrect pass-through)

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/orbital-r-fork#18]
## Summary

The `orbital_keras_dag_impl()` function dispatches `BatchNormalization` / `batch_normalization` layers as **pass-throughs** — it copies the inbound expressions unchanged. This is incorrect: at inference time, BatchNorm applies a deterministic affine transformation using the layer's running mean, variance, gamma, and beta:

```
y_i = ((x_i - running_mean_i) / sqrt(running_var_i + eps)) * gamma_i + beta_i
```

All four parameter vectors are stored in the model weights and are fixed at inference time, making this fully expressible as SQL.

## Acceptance Criteria

- `BatchNormalization` layer case in `orbital_keras_dag_impl()` changed from pass-through to proper normalization
- Retrieve per-feature parameters from the keras layer object: `gamma`, `beta`, `moving_mean`, `moving_variance`
- `epsilon` attribute retrieved via `layer$epsilon`
- The normalization formula applied element-wise across the expression vector
- Output stored in the registry under the layer name
- Integration test: `Sequential(Dense(relu), BatchNormalization, Dense)` — verify orbital output matches `predict(model, df)` (skip if keras3 not installed)

## Notes

- keras3 API: `layer$gamma$numpy()`, `layer$beta$numpy()`, `layer$moving_mean$numpy()`, `layer$moving_variance$numpy()`; also obtainable via `layer$get_weights()` (order: gamma, beta, moving_mean, moving_variance)
- `build_mlp_pre_act()` in `ann-helpers.R` uses `kernel %*% x + bias` pattern — BatchNorm should slot in between that and `activation_expr()`
- For parity: Python `BatchNormalizationTranslator` tracked in orbital-1 issue #17

**Comment by davidrsch** [comment]
Implemented in commit `46d7140` on `feat/ann-mlp-support`.\n\nChanges in `R/model-keras.R` `orbital_keras_dag_impl()`:\n- **Fixed weight mapping** — Now uses `l$get_weights()` per layer instead of positional global indexing. The previous `all_weights[[2*dense_count-1]]` indexing was wrong whenever non-Dense layers (BatchNorm, PReLU) contributed to the weight list.\n- **BatchNormalization dispatch**: reads `gamma`, `beta`, `moving_mean`, `moving_variance` via `l$get_weights()`; reads `epsilon` via `l$epsilon`; applies `((x - mean) / sqrt(var + eps)) * gamma + beta` element-wise; stores intermediate expressions under `orbital_bn_{lname}_h{i}` names.\n- **DAG trigger** in `orbital_keras_impl()` updated to route `batchnorm` layers to the DAG path.\n\nIntegration test added: `keras3 Functional model with BatchNormalization translates correctly` — verifies `orbital_bn_` intermediates appear in output and `.pred` is produced.
